src/main/scala/com/databricks/labs/mosaic/core/index/BNGIndexSystem.scala

```diff
@@ -368,16 +476,22 @@ object BNGIndexSystem extends IndexSystem with Serializable {
       */
     override def getResolution(res: Any): Int = {
```

Comment:
As per the slack discussion, need to make sure that all expressions that compute indexes use the getResolution method for the correct index system (PointIndexGeom is still using H3).

Reply:
fixed in the last commit.